### DISCUSSION TITLE
Support for discriminated form?

### DISCUSSION THREAD

**Discussion by yatesco** [tokio-rs/axum#1221] [Q&A] [answered]
I'm expecting the answer is no, and I understand why, but on the off chance ;-)....

I have a front end which posts to an `/action` endpoint, but the body of that action can have multiple structures. Is it possible to intelligently deserialise a form to an enum? Are there any examples?

I'm talking about something like:
```rust
enum MyForm {
  Rename(String),
  Delete.
  SomethingElse(String, usize)
}
```

with POSTS like `action=rename&rename[0]="something"`, or `action=delete` etc.

I think I'm going to end up JSONing it on the front end but I'm using the excellent https://htmx.org so would prefer to avoid that if possible.

**Comment by davidpdrsn** [comment, marked as answer]
Almost. This should work

```rust
#[derive(Deserialize, Debug)]
#[serde(tag = "action", rename_all = "kebab-case")]
enum Input {
    Rename { rename: String },
    Delete,
    SomethingElse { foo: String, bar: usize },
}
```

Although you'll have to POST the data as `action=rename&rename=foobar`.

There might be some serde attributes that you can use to get exactly what you wrote but I'm not sure.

**Reply by yatesco** [reply]
thank you!
